templates/client/html/catalog/detail/body.php

```diff
@@ -48,7 +48,23 @@
 
 
 ?>
-<?php if( isset( $this->detailProductItem ) ) : ?>
+<?php if( isset( $this->detailProductItem ) ) : 
+
+        // preselection of a variant product in a selection article
+        $preSelectVariantAttributes = [];
+        if ($this->param( 'productId' ) && is_numeric($this->param( 'productId' ))):
```

Comment:
The product ID doesn't has to be numeric, that's only currently the case in relational databases